### DISCUSSION TITLE
We got duplicated Child_SAs after running IPSec tunnels between Strongswan CVM and FortiGate firewall

### DISCUSSION THREAD

**Discussion by jerryishe** [strongswan/strongswan#1001] [Q&A]
Hello Team,
We use strongswan to create IPSec tunnel between CentOS 7.3 VM and Fortigate 600D firewall, and it goes well.
However, after running for 2days, something wired thing happened and it confused us.
We set reauth_time as 2days 172800s (two days), and after reauthing, the connection goes good, IPSec tunnel keeps UP at fortiGate end. **But we see the old Child_SA is still existed, not deleted**. Traffic is on the new Child_SA so the connection goes good.
That means after reauthing, the old Child_SA is failed to be deleted? 
How can I delete this duplicated ChildSA? Now I found that if I systemctl restart the strongswan, the duplicated Child_SA will be deleted. But that is not a good way…………
 
We set IPSec session between these two devices with below attachments:
1.  strongswan configuration and some swanctl message
[strongswan_configuration.log](https://github.com/strongswan/strongswan/files/8521338/strongswan_configuration.log)
2. fortigate 600D configuration
[FortiGate_configuration.log](https://github.com/strongswan/strongswan/files/8521339/FortiGate_configuration.log)
3. strongswan log message(too much dpd log because duplicated Child_SAs) You could see line 50142-50164, 50642-50683, 63758-63798 to get reauth/rekeying log
[strongswan_var_message.log](https://github.com/strongswan/strongswan/files/8521340/strongswan_var_message.log)

Hope you could give me a good solution. Thanks.

Jerry.

**Comment by tobiasbrunner** [comment]
Your log doesn't show how the duplicates were created in the first place. But it's possible that it's the other peer that does so. You could try enabling _make-before-break_ reauthentication (the other peer might create a duplicate IKE_SA if strongSwan first deletes the IKE_SA before it creates a new one, instead of doing that overlapping), or you could disable reauthentication altogether and just use rekeying. See [here](https://docs.strongswan.org/strongswan-docs/5.9/config/rekeying.html) for details.

**Comment by bitroniq** [comment]
I have the same issue with the latest FortiGate firmware.

Trying now to use `make_before_break = yes`

**Reply by bitroniq** [reply]
Unfortunately, that doesn't help. Re-authentication causes the tunnel to stuck.

**Reply by bitroniq** [reply]
@jerryishe have you resolved this problem with FortiGate?
